From 29ed309e0bf2613a7db358eee8b9975f50884a14 Mon Sep 17 00:00:00 2001
From: surfacebot <surfacebot@users.noreply.github.com>
Date: Tue, 23 Jun 2026 14:33:16 +0200
Subject: [PATCH] Remove stupid merge leftover

---
 drivers/scsi/sd.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/scsi/sd.c b/drivers/scsi/sd.c
index 42378a7b04a28..ec52c3a4fa14e 100644
--- a/drivers/scsi/sd.c
+++ b/drivers/scsi/sd.c
@@ -119,7 +119,6 @@ static const char *sd_cache_types[] = {
 	"write back, no read (daft)"
 };
 
-<<<<<<< HEAD
 static int sd_large_pool_create(void)
 {
 	mutex_lock(&sd_mutex_lock);
-- 
2.52.0

